Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -177,14 +178,32 @@ void WebCookieJar::setCookies(WebCore::Document& document, const URL& url, const
     WebProcess::singleton().ensureNetworkProcessConnection().connection().send(Messages::NetworkConnectionToWebProcess::SetCookiesFromDOM(document.firstPartyForCookies(), sameSiteInfo, url, frameID, pageID, applyTrackingPreventionInNetworkProcess, cookieString, shouldRelaxThirdPartyCookieBlocking(webFrame)), 0);
 }
 
-void WebCookieJar::cookiesAdded(const String& host, const Vector<WebCore::Cookie>& cookies)
+void WebCookieJar::cookiesAdded(const String& host, Vector<WebCore::Cookie>&& cookies)
 {
-    m_cache.cookiesAdded(host, cookies);
+    auto cookieChangeListenerIter = m_cookieChangeListeners.find(host);
+    if (cookieChangeListenerIter != m_cookieChangeListeners.end())
+        return;
+
+    auto cookieChangeListener = cookieChangeListenerIter->value;
+    if (!cookieChangeListener)
+        return;
+
+    for (auto& cookie : cookies)
+        cookieChangeListener->addCookie(host, WTFMove(cookie));
```

Comment:
Why are we now passing the cookies to the observer one by one instead of passing the full vector like we used to do. Seems less efficient.